### PR TITLE
Upgrade to haskell-lsp 0.15

### PR DESCRIPTION
**Pull request by lorenzo**

### CODE REVIEW

File: hie-plugin-api/Haskell/Ide/Engine/Ghc.hs [hunk 1]
```diff
@@ -45,10 +47,23 @@ import           GHC
...
 
-type Diagnostics = Map.Map Uri (Set.Set Diagnostic)
+newtype Diagnostics = Diagnostics (Map.Map NormalizedUri (Set.Set Diagnostic))
+  deriving (Show, Eq)
+
+instance Semigroup Diagnostics where
+  Diagnostics d1 <> Diagnostics d2 = Diagnostics (d1 <> d2)
+
+instance Monoid Diagnostics where
+  mempty = Diagnostics mempty
```
Comment (bubba) [inline hie-plugin-api/Haskell/Ide/Engine/Ghc.hs]: Can these be derived?

Comment (wz1000) [reply hie-plugin-api/Haskell/Ide/Engine/Ghc.hs]: The default monoid instance for Map is probably not what we want. This should be Map.Union with Set.union. the default semi group operation is leftbiased union.

Comment (bubba) [reply hie-plugin-api/Haskell/Ide/Engine/Ghc.hs]: In any case I think `ToJSON` should be derivable. Although it looks like `NormalizedUri` has no instance for `ToJSON/FromJSON`. This should probably be added in haskell-lsp.

**Comment by lorenzo** [comment]
I'll merge now, please let me know if you wanted me to take any extra actions and I'll open another pull request.

**Pull request merged**